### ISSUE TITLE
Indentation: incorrect expectation for wrapped arguments of chained calls

### ISSUE THREAD

**Issue by eric-milles** [checkstyle/checkstyle#3208]
We are seeing warnings for wrapped lines of our builder calls. For eaxmple:
```
    public static final ResultType WHATEVER = ResultType.builder("WHATEVER") //$NON-NLS-1$
        .passThrough(true)
        .contentType("WHATEVER") //$NON-NLS-1$
        .supportedMetadataTasks(
            TaskOne.RESULT,
            TaskTwo.RESULT,
            TaskThree.RESULT)
        .build();
```
This results in warnings for each line of the varargs method call.  The warning is seeking the lines to be indented at the same level as the methods (one less tab each).

We use the indentation check in this way:
```
        <module name="Indentation">
            <message key="indentation.error" value="Line should be indented by {2} spaces" />
            <message key="indentation.child.error" value="Line should be indented by {2} spaces" />
            <message key="indentation.error.multi" value="Line should be indented by one of: {2}" />
            <property name="forceStrictCondition" value="true" />
            <property name="basicOffset" value="4" />
            <property name="caseIndent" value="0" />
        </module>
```

**Comment by kevinconaway** [comment]
@rnveach I'm seeing something similar as well when working with Apache Spark.  I've created a small test class that exposes this issue:

```java

public class Test {
    
    public static void main(String [] args) {
        MockRDD mockRDD = new MockRDD();
        mockRDD.mapToPair(
            new ToSequenceFilePair()
        ).partitionBy(
            new TimeBasedPartitioner(partitions)
        ).saveAsHadoopFile(
            outputPath.toString(),
            LongWritable.class,
            BytesWritable.class,
            PartitionedOutputFormat.class,
            Lz4Codec.class
        );
    }
    
    private static class MockRDD {

        public MockRDD mapToPair(Object arg) {
            return this;
        }

        public MockRDD partitionBy(Object arg) {
            return this;
        }

        public MockRDD saveAsHadoopFile(Object... args) {
            return this;
        }

    }
    
}
```

The checkstyle rule config:

```xml
<module name = "Checker">
    
    <module name="TreeWalker">
        <module name="Indentation"/>
    </module>
    
</module>
```

Running this on the CLI gives me:

```
java -jar checkstyle-6.19-all.jar -c checkstyle.xml Test.java
Starting audit...
[ERROR] /Users/kconawa/Downloads/Test.java:12: 'method call' child have incorrect indentation level 12, expected level should be 16. [Indentation]
[ERROR] /Users/kconawa/Downloads/Test.java:13: 'method call' child have incorrect indentation level 12, expected level should be 16. [Indentation]
[ERROR] /Users/kconawa/Downloads/Test.java:14: 'method call' child have incorrect indentation level 12, expected level should be 16. [Indentation]
[ERROR] /Users/kconawa/Downloads/Test.java:15: 'method call' child have incorrect indentation level 12, expected level should be 16. [Indentation]
[ERROR] /Users/kconawa/Downloads/Test.java:16: 'method call rparen' have incorrect indentation level 8, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 5 errors.
```

If I change the code to the indentation below it passes but it is clearly incorrect:

```java
        mockRDD.mapToPair(
            new ToSequenceFilePair()
        ).partitionBy(
            new TimeBasedPartitioner(partitions)
        ).saveAsHadoopFile(
            outputPath.toString(),
                LongWritable.class,
                BytesWritable.class,
                PartitionedOutputFormat.class,
                Lz4Codec.class
            );
```

**Comment by romani** [comment]
@rnveach , really weird behaviour now, definitely a bug.

**Comment by rnveach** [comment]
Yes. Output has not changed since before 5.9, so this has been around for a while.

Issue is most likely because we call the `LineWrapping` on the entire method call: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/MethodCallHandler.java#L166
I will have to see how each IDE handles it by default.

**Comment by rnveach** [comment]
Code from PR.
````
        MockRDD mockRDD = new MockRDD(); //indent:8 exp:8
        mockRDD.mapToPair( //indent:8 exp:8
            null //indent:12 exp:12
        ).saveAsHadoopFile(  //indent:8 exp:8
            null //indent:12 exp:12
        ); //indent:8 exp:8
````
All IDEs indent the given code the same as commented by default.

When the `(` is line separated from `saveAsHadoopFile`, Eclipse keeps the `(` at indent 4, while the other IDEs indent it to 8. Children are indented relative to parent.
**Eclipse view**:
````
		).saveAsHadoopFile//
		( // indent:8 exp:8
				null, // indent:12 exp:12
		); // indent:8 exp:8
````
**Other IDEs view**:
````
        ).saveAsHadoopFile//
                ( //indent:8 exp:8
                        null //indent:12 exp:12
                );  //indent:8 exp:8
````
When the `)` is line separated from `.saveAsHadoopFile`, all IDEs indent `.saveAsHadoopFile` to 8 and keep children relative to parent. Eclipse seems to be the only one that keeps the final `);` at indent 4 no matter what, even when previous lines are increased in indent.
**Eclipse view**:
````
		)//
				.saveAsHadoopFile( // indent:8 exp:8
						null, // indent:12 exp:12
		); // indent:8 exp:8
````
**Other IDEs view**:
````
        )//
                .saveAsHadoopFile( //indent:8 exp:8
                        null //indent:12 exp:12
                );  //indent:8 exp:8
````

**Comment by romani** [comment]
Fix is merged

**Comment by mageddo** [comment]
I'm having this exactly bug. But I'm using the latest version

**Comment by rnveach** [comment]
@mageddo Please open a new issue with all the details.
